### ISSUE TITLE
Scaling max/min is applied each time with disabled autoscaling

### ISSUE THREAD

**Issue by aeimer** [mongodb/terraform-provider-mongodbatlas#482]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v0.15.3
on darwin_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.63.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/mongodb/mongodbatlas v0.9.1
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_cluster" "main" {
  for_each = local.mongodb_clusters

  name       = each.value
  project_id = mongodbatlas_project.main.id

  auto_scaling_compute_enabled                    = var.mongodbatlas_config[each.value].mongodbatlas_autoscale # Set to false
  auto_scaling_compute_scale_down_enabled         = var.mongodbatlas_config[each.value].mongodbatlas_auto_down_scale # Set to false
  auto_scaling_disk_gb_enabled                    = true
  cluster_type                                    = "REPLICASET"
  mongo_db_major_version                          = "4.4"
  provider_auto_scaling_compute_max_instance_size = var.mongodbatlas_config[each.value].mongodbatlas_instance_max # Set to M10
  provider_auto_scaling_compute_min_instance_size = var.mongodbatlas_config[each.value].mongodbatlas_instance_min # Set to M10
  provider_backup_enabled                         = true
  provider_instance_size_name                     = var.mongodbatlas_config[each.value].mongodbatlas_instance_min
  provider_name                                   = "AZURE"

  advanced_configuration {
    minimum_enabled_tls_protocol = "TLS1_2"
  }

  replication_specs {
    num_shards = 1
    dynamic "regions_config" {
      for_each = local.mongodb_regions
      content {
        region_name     = regions_config.value["atlasregion"]
        electable_nodes = regions_config.value["electable_nodes"]
        priority        = regions_config.value["priority"]
      }
    }
  }

  labels {
    key   = "environment"
    value = var.environment_name
  }

  lifecycle {
    ignore_changes = [
      provider_instance_size_name
    ]
  }
}

```

### Expected Behavior

The provider should not find any changes to be done.
Especially as the update needs quite a few seconds to minutes, it would be nice to be skipped.

### Actual Behavior

The provider show `provider_auto_scaling_compute_max_instance_size` and `provider_auto_scaling_compute_min_instance_size` to be `not set`?!.

```bash
$ tf plan
  # mongodbatlas_cluster.main["sandbox-main"] will be updated in-place
  ~ resource "mongodbatlas_cluster" "main" {
        id                                              = "someid"
        name                                            = "sandbox-main"
      + provider_auto_scaling_compute_max_instance_size = "M10"
      + provider_auto_scaling_compute_min_instance_size = "M10"
        # (27 unchanged attributes hidden)
        # (4 unchanged blocks hidden)
    }

$ tf apply
mongodbatlas_cluster.main["sandbox-main"]: Modifying... [id=someid]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 10s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 20s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 30s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 40s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 50s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 1m0s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 1m10s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 1m20s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Still modifying... [id=someid, 1m30s elapsed]
mongodbatlas_cluster.main["sandbox-main"]: Modifications complete after 1m39s [id=someid]
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

**Comment by themantissa** [comment]
@aeimer do you happen to have debug logs as requested in our issue template?  Otherwise we'll have to find time to repro so logs speed up the process.  @nikhil-mongo

**Comment by aeimer** [comment]
I'll gather that info on Monday and send it to you

Thanks

**Comment by aeimer** [comment]
@themantissa The log is sooo long and there are things in it like access keys. So I actually can't publish the log... Are there parts that are interesting for you? I may can copy them for you and clean them manually.

**Comment by themantissa** [comment]
@aeimer the logs can indeed be redacted (remove any secrets). The most important sections are when Terraform is sending the actual data to the API and refreshing so we can see what is happening specifically.

**Comment by nikhil-mongo** [comment]
@aeimer and @themantissa 
I can repro this what is mentioned in this bug ticket.
When the `auto_scaling_compute_enabled` and `auto_scaling_compute_scale_down_enabled` are set to false, terraform provider will ignore the values set for `provider_auto_scaling_compute_max_instance_size` and `provider_auto_scaling_compute_min_instance_size` which is as expected. 

```
021-06-29T10:52:06.258+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:   "autoScaling": {
2021-06-29T10:52:06.258+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:    "compute": {
2021-06-29T10:52:06.258+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:     "maxInstanceSize": null,
2021-06-29T10:52:06.258+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:     "minInstanceSize": null
2021-06-29T10:52:06.258+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:    }
2021-06-29T10:52:06.258+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:   },
```
But when the  value of `auto_scaling_compute_enabled` and `auto_scaling_compute_scale_down_enabled` are set to true then the values set for `provider_auto_scaling_compute_max_instance_size` and `provider_auto_scaling_compute_min_instance_size` are set as expected.

```
2021-06-29T10:55:17.650+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:   "autoScaling": {
2021-06-29T10:55:17.650+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:    "compute": {
2021-06-29T10:55:17.650+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:     "maxInstanceSize": "M20",
2021-06-29T10:55:17.650+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:     "minInstanceSize": "M10"
2021-06-29T10:55:17.650+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:    }
2021-06-29T10:55:17.650+0530 [DEBUG] provider.terraform-provider-mongodbatlas_v0.9.1:   },
```

- We need to understand here that whenever we run terraform plan or terraform apply, terraform refreshes the state file and matches the configuration the state file. Since, the value for  `provider_auto_scaling_compute_max_instance_size` and `provider_auto_scaling_compute_min_instance_size` is not set in the terraform.tfstate file, terraform attempts to update it and shows as updating but never updates it due to the leading parameter of `auto_scaling_compute_enabled` and `auto_scaling_compute_scale_down_enabled`  set to false.

```
"provider_auto_scaling_compute_max_instance_size": "",
"provider_auto_scaling_compute_min_instance_size": "",
```

- Below are the logs with the values set to `true` or `false`.

[trace-false.log](https://github.com/mongodb/terraform-provider-mongodbatlas/files/6730889/trace-false.log)
[trace-true.log](https://github.com/mongodb/terraform-provider-mongodbatlas/files/6730890/trace-true.log)

**Comment by aeimer** [comment]
Cools thanks. Do you need my logs then? Can I help somehow?

**Comment by nikhil-mongo** [comment]
@aeimer No, we have the logs now. Let us look into it now and see if how we want to take it forward. The provider is behaving how it should but lets see if there are chances to make changes.

**Comment by themantissa** [comment]
@nikhil-mongo thank you so much for getting that done and saving everyone some stress and time!   We should be able to improve the handling of that in the provider.  Can you put it in a PRODTRIAGE?  Thank you!

**Comment by nikhil-mongo** [comment]
PRODTRIAGE-1824

**Comment by themantissa** [comment]
Addressed in 1.0.1
